### PR TITLE
Clockcult updates

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Does some updates to clockcult.

 - There is no longer a hard conversion limit. Instead, once 6 players have been converted an 80 minute timer starts. This timer is reduced every time somebody is converted. You can no longer just keep people in a prison until the gateway opens, as there is no point at which the gateway will suddenly open. Additionally, speedy cults will be rewarded with extra time and therefor more converts.
 - Teleporting now cancels any activated scriptures. You can no longer teleport and immediately stun somebody.
 - Regular armour has been significantly buffed against projectiles and melee attacks.
 - Most other forms of armour have been buffed.
 - Almost all clockcult weapons have been nerfed, including weaker damage away from the ark.
 - Clockcult warp is quieter
 - Vanguard now gives you immortality until it wears off, when you will suddenly die if you took too much damage
 - Sentinel's compromise now stuns you on use

## Why It's Good For The Game

Removes the hard conversion limit and replaces it with a softer conversion limit that prevents cultists from being able to imprison people until the end.
Punishes the cult for killing station members and giving them a boost if the station is heavilly mindshielded.
No more teleporting onto someone and instantly stunning them.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image
image
image

</details>

## Changelog
:cl:
tweak: Clockcult now has a timer that starts once 6 people have been converted.
tweak: The clockcult timer will be reduced every time a player is converted.
tweak: Clockcult teleportation cancels active scriptures
tweak: Clockcult gain power efficiency for every applied mindshield.
tweak: Clockcult lose power efficiency for every dead crewmember.
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
> tweak: The clockcult timer will be reduced every time a player is converted.

And give the time back (or half of it) when deconverted?

**Comment by heepox** [comment]
Just convert 5 and imprison the rest.
Just kill absolutely everyone so it wont matter (but what if they die to non clock cult things and cult gets punished?)
just mindshield all cultists since it wont deconvert them.
back to stealing stunbatons and using bolas it is.

Team antag so number 1 propably wont happen
number 2 is easy with a flood or fusion or just being robust
Number 3 easy as pie.
Number 4 The Greytide cult but they can teleport aswell as having insuls and tools.

Good idea... will see if it works...

**Comment by PowerfulBacon** [comment]
> Just convert 5 and imprison the rest.

Well you can do that, but keep in mind you won't have the man power of cultists during this time. Mindshielding cultists is a valid strategy and makes the mindshields more valuable.
The deaths only have a minor impact, so 1 person killing themselves won't destroy the cult (Plus 1 person dying is replaced by another being mindshielded anyway).

**Review by Archanial — DISMISSED** [review]

**Comment by PowerfulBacon** [comment]
Line 66 in replica_fabricator.dm has a divsion by 0 error

File: code/modules/antagonists/clock_cult/helpers/clockwork_cost_multiplier.dm [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+		return 1
+	var/baseline_power = GLOB.joined_player_list.len
+	var/power = 0
+	for(var/mob/living/carbon/human/player in GLOB.mob_living_list)
+		//Only count alive players that joined the game at roundstart
+		if(player.stat != DEAD && (player in GLOB.joined_player_list))
+			power ++
+			if(HAS_TRAIT(player, TRAIT_MINDSHIELD))
+				power ++
+	. = baseline_power / power
```
Comment (PowerfulBacon) [inline code/modules/antagonists/clock_cult/helpers/clockwork_cost_multiplier.dm]: Division by 0 exception

**Review by PowerfulBacon — COMMENTED** [review]
The conversion limit needs to be raised

**Review by PowerfulBacon — COMMENTED** [review]
[14:24:59] Runtime in clockwork_conversion_tracker.dm, line 83: Cannot read null.mind

**Review by PowerfulBacon — COMMENTED** [review]
Timer goes negative when it finishes

**Comment by PowerfulBacon** [comment]
Clockcult should be blocked from building reinforced walls on cogland

**Comment by PowerfulBacon** [comment]
image

**Comment by PowerfulBacon** [comment]
image

**Comment by PowerfulBacon** [comment]
image

**Comment by PowerfulBacon** [comment]
image

**Comment by PowerfulBacon** [comment]
image

**Comment by PowerfulBacon** [comment]
Sentinal's compromise is extremely powerful. Also the suits have way too much melee resistance

**Comment by PowerfulBacon** [comment]
Toxin damage on sentinel's compromise should be forced.

**Comment by PowerfulBacon** [comment]
image

Scarabs should build faster, power efficiency is bugged

**Comment by Fronsis** [comment]
Armor value of clockies might be too high? On recent test-merged clockie round i shot him 5-6 times with shotgun slug ammo and he didn't go down

He was also moving quite fast IMO if they have a buffed up armor they should be at riot-armor speed? to me he seemed like he shrugged off the shots and managed to run away quickly, imo if the armor gives a reasonable slow that would prevent them to go full armor into the station and start hunting sec/heads (As it happened to me)  but would still allow them to defend the reebee since most cases of front-line fighting they have plenty of cover and there's no much need to run.

Also what's the visual indicator of Vanguard? If it gives immortality until you take too much damage they might use it to rush and battle on station + reebee

**Comment by PowerfulBacon** [comment]
The armour has very high slowdown and slugs have insanely bad armour penetration (doing almost no damage against armoured targets) but I agree that the bullet armour is too high.
It could have been someone wearing the speed robes which let you run at very high speeds, but have weaker protection than the other armours combined with the fact that slugs aren't effective against armoured targets.

Vanguard visual indicator is to show that the ability to active, it provides 30 seconds of stun immunity and death immunity until it wears off, which is a pretty short time for someone to run into battle on the station (Since they then have to return to reebe and heal all their damage before the ability wears off)

**Comment by PowerfulBacon** [comment]
Good PR, stale

**Pull request closed without merging**